controllers/actions.github.com/autoscalingrunnerset_controller.go

```diff
@@ -338,17 +347,59 @@ func (r *AutoscalingRunnerSetReconciler) createRunnerScaleSet(ctx context.Contex
 		autoscalingRunnerSet.Annotations = map[string]string{}
 	}
 
-	autoscalingRunnerSet.Annotations[runnerScaleSetIdKey] = strconv.Itoa(runnerScaleSet.Id)
-	logger.Info("Adding runner scale set ID as an annotation")
-	if err := r.Update(ctx, autoscalingRunnerSet); err != nil {
-		logger.Error(err, "Failed to add runner scale set ID")
+	logger.Info("Adding runner scale set ID and runner group name as an annotation")
+	err = patch(ctx, r.Client, autoscalingRunnerSet, func(obj *v1alpha1.AutoscalingRunnerSet) {
+		obj.Annotations[runnerScaleSetIdKey] = strconv.Itoa(runnerScaleSet.Id)
+		obj.Annotations[runnerScaleSetRunnerGroupNameKey] = runnerScaleSet.RunnerGroupName
+	})
+
+	if err != nil && !kerrors.IsNotFound(err) {
```

Comment:
Maybe pull in the case where this is not found to report it as such. Because if err is not found, then we should not log that we updated with runner scale set id as an annotation

Reply:
updated. 🙇